### ISSUE TITLE
compile_file claims compilation failed when it succeeded

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#3512]
Imported from SourceForge on 2024-07-08 07:09:56
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2005-01-17 08:19:17
Original: https://sourceforge.net/p/maxima/bugs/669

---

compile\_file uses the tertiary return value of
COMPILE-FILE to tell whether the compilation succeeded.
At least one lisp \(Clisp\) returns a positive value if
there are warnings, so compile\_file thinks the
compilation failed when actually there weren't errors. 

A comment in the code \(src/transs.lisp\) says

;; If the compiler encountered errors, don't set
bin-file to
;; indicate that we found errors. Is this what we want?

I think the answer is "no".

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-08 07:09:58
Created by **[demoss](https://sourceforge.net/u/demoss/)** on 2005-03-30 13:15:54
Original: https://sourceforge.net/p/maxima/bugs/669/#d393

---

Logged In: YES 
user\_id=283348

The third return value of COMPILE-FILE indicates presence of errors or 
non-style-warning warnings; non-style-warning warnings are an indication of 
real problems and undefined behaviour, and should be treated as such: eg. 
SBCL emits full warnings for unbound variables and type-errors detected at 
compile-time.

In other words, the current behaviour is IMO good and safe.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-08 07:10:01
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2006-04-10 18:47:51
Original: https://sourceforge.net/p/maxima/bugs/669/#fb63

---

- **labels**: 460522 --> Lisp Core - Translator

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-08 07:10:05
Created by **[crategus](https://sourceforge.net/u/crategus/)** on 2010-09-12 11:49:56
Original: https://sourceforge.net/p/maxima/bugs/669/#af9e

---

From the postings of this bug report, there seems to be no consent what should be the correct and save behavior of compile\_file. Furthermore, we have no example which shows a problem with the current behavior of the function compile\_file. Because, this is a very old bug report, there seems to be no really problems. I suggest to close this bug report as 'won't fix'.

Setting the status to pending.
Dieter Kaiser

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-08 07:10:08
Created by **[crategus](https://sourceforge.net/u/crategus/)** on 2010-09-12 11:49:56
Original: https://sourceforge.net/p/maxima/bugs/669/#12b9

---

- **status**: open --> pending
